tests/test_pipeline.py

```diff
@@ -0,0 +1,93 @@
+import signal
+import threading
+import time
+
+import actfw_core
+from actfw_core.task import Consumer, Join, Pipe, Producer, Tee
+
+
+class Counter(Producer):
+    def __init__(self):
+        super(Producer, self).__init__()
+        self.n = 0
+
+    def proc(self, _):
+        time.sleep(0.01)
+        n = self.n
+        self.n += 1
+        return n
+
+
+class Incrementer(Pipe):
+    def __init__(self):
+        super(Incrementer, self).__init__()
+
+    def proc(self, x):
+        return x + 1
+
+
+class Adder(Pipe):
+    def __init__(self):
+        super(Adder, self).__init__()
+
+    def proc(self, xs):
+        return sum(xs)
+
+
+class Printer(Consumer):
+    def __init__(self):
+        super(Printer, self).__init__()
+        self.xs = []
+
+    @property
+    def logs(self):
+        return self.xs
+
+    def proc(self, x):
+        self.xs.append(x)
+        # print(x)
```

Comment:
Deletable?

Reply:
What do you think about uncommenting this line? If "not noizy", I'll uncomment it.

Reply:
I think it's noisy because it affects the output of `nose2`.
[nose2 provides buffering of outputs](https://docs.nose2.io/en/latest/plugins/buffer.html) but it doesn't "no output" AFAIK.

But it's OK to let it commented-out. I intended to notice "if you forgot to delete before commit".

Reply:
Renaming `Printer` (without `print`) to `Logger` by https://github.com/Idein/actfw-core/pull/15/commits/b5674eaeb899625f7aab2759b219411edd83822b makes this point sound natural.